ports/vcpkg-make/vcpkg_make_common.cmake

```
@@ -0,0 +1,659 @@
+        elseif(arg MATCHES "^-(isysroot|m32|m64|m?[Aa][Rr][Cc][Hh]|target)\$")
+            vcpkg_list(SET pattern "${arg}")
+            continue()
+        else()
+            continue()
+        endif()
+        vcpkg_list(APPEND ABIFLAGS "${pattern}")
+        list(REMOVE_ITEM CFLAGS "${pattern}")
+        list(REMOVE_ITEM CXXFLAGS "${pattern}")
+        list(REMOVE_ITEM LDFLAGS "${pattern}")
+        set(pattern "")
+    endforeach()
+
+    # Filter common CPPFLAGS out of CFLAGS and CXXFLAGS
+    if(NOT arg_DISABLE_CPPFLAGS)
+        set(CPPFLAGS "")
+        set(pattern "")
+        foreach(arg IN LISTS CXXFLAGS)
+            if(NOT pattern STREQUAL "")
+                vcpkg_list(APPEND pattern "${arg}")
+            elseif(arg MATCHES "^-(D|isystem).")
+                vcpkg_list(SET pattern "${arg}")
+            elseif(arg MATCHES "^-(D|isystem)\$")
+                vcpkg_list(SET pattern "${arg}")
+                continue()
+            else()
+                continue()
+            endif()
+            string(FIND "${CFLAGS}" "${pattern} " index)
+            if(NOT index STREQUAL "-1")
+                vcpkg_list(APPEND CPPFLAGS ${pattern})
+                list(REMOVE_ITEM CFLAGS "${pattern}")
+                list(REMOVE_ITEM CXXFLAGS "${pattern}")
+                list(REMOVE_ITEM LDFLAGS "${pattern}")
+            endif()
+        endforeach()
+        set(pattern "")
+        foreach(arg IN LISTS CFLAGS)
+            if(NOT pattern STREQUAL "")
+                vcpkg_list(APPEND pattern "${arg}")
+            elseif(arg MATCHES "^-(D|isystem)\$")
+                vcpkg_list(SET pattern "${arg}")
+                continue()
+            elseif(arg MATCHES "^-(D|isystem).")
+                vcpkg_list(SET pattern "${arg}")
+            else()
+                continue()
+            endif()
+            string(FIND "${CXXFLAGS}" "${pattern} " index)
+            if(NOT index STREQUAL "-1")
+                vcpkg_list(APPEND CPPFLAGS ${pattern})
+                list(REMOVE_ITEM CFLAGS "${pattern}")
+                list(REMOVE_ITEM CXXFLAGS "${pattern}")
+                list(REMOVE_ITEM LDFLAGS "${pattern}")
+            endif()
+            vcpkg_list(SET pattern)
+        endforeach()
+    endif()
+
+    # libtool tries to filter CFLAGS passed to the link stage via a allow list.
+
+    # This approach is flawed since it fails to pass flags unknown to libtool
+    # but required for linking to the link stage (e.g. -fsanitize=<x>).
+    # libtool has an -R option so we need to guard against -RTC by using -Xcompiler.
+    # While configuring there might be a lot of unknown compiler option warnings
+    # due to that; just ignore them.
+    set(compiler_flag_escape "")
+    if(arg_COMPILER_FRONTEND STREQUAL "MSVC" AND NOT arg_NO_FLAG_ESCAPING)
+        set(compiler_flag_escape "-Xcompiler") # TODO: Check why this had a trailing space? We are using lists so it shouldn't be necessary here

```

```suggestion
        set(compiler_flag_escape "-Xcompiler")
```

It doesn't have a trailing space now.